### PR TITLE
Add extra docker checks

### PR DESCRIPTION
**Pull request by ciaranRoche**
Currently we only check the version of docker, this change adds two checks, one to see if docker exists, and the other to see if docker is currently running.

@aidenkeating

### CODE REVIEW

File: installer/clean [hunk 1]
```diff
@@ -0,0 +1,22 @@
+#!/bin/sh
```
Comment (wtrocki) [inline installer/clean]: What this file is for? It seem to be unrelated to your PR?

Comment (ciaranRoche) [reply installer/clean]: My bad, I just added all to my commit, I did not edit that file, but I did run the clean.sh file while testing the install.sh file.

Comment (wtrocki) [reply installer/clean]: `git reset --soft head~1` will do the job.

File: installer/install.sh [hunk 2]
```diff
@@ -67,6 +67,22 @@ function check_passed_msg() {
...
+  check_exists_msg "Docker" 
+  command -v docker &>/dev/null
+  docker_exists=${?}; if [[ ${docker_exists} -ne 0 ]]; then
+    does_not_exist_msg "Docker" "https://www.docker.com/get-docker"
+    exit 1
+  fi
+  check_passed_msg "Docker"
+
+  echo -e "Checking Docker is running"
+  command ps -A | grep docker | grep -v grep &>/dev/null
```
Comment (aidenkeating) [inline installer/install.sh]: For me this command will have a return code of 0 when Docker is not running as I have vscode plugins for docker that show up in grep. I wonder if just attempting a `docker ps` would be enough here as it's return code when it cannot connect to the docker daemon is 1? What do you think?
